File: program_data/github_java_sort_function/3/100.java
package companies.apple.experience;

import java.util.*;


public class TopologicalSort {
    
    public static List<List<String>> topSort(Map<String, List<String>> dependencies) {
        Map<String, Integer> inCount = new HashMap<String, Integer>();
        List<List<String>> ret = new ArrayList<List<String>>();
        Set<String> roots = new HashSet<String>();

        for (String n : dependencies.keySet()) {
            if (!inCount.containsKey(n)) {
                roots.add(n);
            }

            for (String s : dependencies.get(n)) {
                if (inCount.containsKey(s)) {
                    inCount.put(s, inCount.get(s) + 1);
                } else {
                    inCount.put(s, 1);
                }

                if (roots.contains(s)) {
                    roots.remove(s);
                }
            }
        }

        ret.add(new ArrayList<String>(roots));
        topSort(inCount, dependencies, ret);

        return ret;
    }

    private static void topSort(Map<String, Integer> inCount, Map<String, List<String>> dependencies, List<List<String>> ret) {
        while (!inCount.isEmpty()) {
            List<String> newNodes = new ArrayList<String>();

            for (String node : ret.get(ret.size() - 1)) {
                if (dependencies.containsKey(node)) {
                    List<String> dependency = dependencies.get(node);
                    for (String n : dependency) {
                        int count = inCount.get(n);
                        inCount.put(n, count - 1);
                        if (count == 1) {
                            newNodes.add(n);
                            inCount.remove(n);
                        }
                    }
                }
            }

            ret.add(newNodes);
        }
    }
}

File: program_data/PKU_raw/25/100.c
int main()
{
	int n, i, j, count = 0;
	cin >> n;
	int a[100];
	for(i = 0; i < 99; i++)
		a[i] = -1;
	a[99] = 1;
	if(n == 0)
		cout << 1;
	else
	{
		for(i = 0; i < n; i++)
		{	
			count = 0;
			for(j = 99; j >= 0; j--)
				if(a[j] != -1)
				{
					count++;
				}
			for(j = 99; j > 99 - count; j--)
				a[j] = a[j] * 2;
			for(j = 99; j > 99 - count; j--)
			{
				if(a[j] >= 10)
				{
					a[j] = a[j] - 10;
					if(a[j - 1] == -1)
					{
						a[j - 1] = 1;
						count++;
						break;
					}
					else 
						a[j - 1] = a[j - 1] + 1; 
				}
			}
		}
	}
	for(i = 99 - count + 1; i <= 99; i++)
		cout << a[i];
	cout << endl;
	return 0;
}


		
		
	
